tree/tree.h

```diff
@@ -15,31 +16,53 @@ class Tree{
   public:
 
     /**
-     * \brief Standard destructor.
+     * Tree class constructor
+     * \param splitter      Splitter object
```

Comment:
Indicate wether the constructed object takes ownership of the pointers (I'm guessing they don't but we have to make it clear).